libs/wingc/src/jsify/code.rs

```diff
@@ -0,0 +1,126 @@
+#[derive(Default)]
+pub struct CodeMaker {
+	lines: Vec<(usize, String)>,
+	indent: usize,
+}
+
+impl CodeMaker {
+	pub fn open<S: Into<String>>(&mut self, line: S) {
+		self.line(line);
+		self.indent += 1;
+	}
+
+	pub fn close<S: Into<String>>(&mut self, line: S) {
+		self.indent -= 1;
+		self.line(line);
+	}
+
+	pub fn line<S: Into<String>>(&mut self, line: S) {
+		self.lines.push((self.indent, line.into()));
+	}
+
+	pub fn add_lines<S: Into<String>>(&mut self, lines: Vec<S>) {
+		for line in lines {
+			self.line(line);
+		}
+	}
+
+	#[allow(dead_code)]
+	pub fn add_code(&mut self, code: CodeMaker) {
+		assert_eq!(code.indent, 0, "Cannot add code with indent");
+		for (indent, line) in code.lines {
+			self.lines.push((indent + self.indent, line));
+		}
+	}
+
+	#[allow(dead_code)]
+	pub fn unindent(&mut self) {
+		self.indent -= 1;
+	}
+
+	pub fn indent(&mut self) {
+		self.indent += 1;
+	}
+}
+
+impl ToString for CodeMaker {
+	fn to_string(&self) -> String {
+		let mut code = String::new();
+		for (indent, line) in &self.lines {
+			code.push_str(&"\t".repeat(*indent));
```

Comment:
Use two spaces instead of tab, which is the common style in JavaScript

Reply:
OK